db-engine/src/main/java/com/splicemachine/db/iapi/store/access/StoreCostController.java

```diff
@@ -120,9 +123,10 @@
      * @throws StandardException Standard exception policy.
      * @see RowUtil
      */
-    void getFetchFromFullKeyCost(BitSet validColumns,
+    void getFetchFromFullKeyCost(boolean forExprIndex,
```

Comment:
That's a good idea. I did it in the current way because when an index on expressions is not covering, we need both base table statistics and index on expressions statistics at the same time. That's why we have them together in `StoreCostController` and have this flag in methods. It's a quick step.

Ideally we should be able to have multiple `StoreCostController`s representing statistics on different conglomerates. A bit more refactoring is needed to achieve it, though. I think when we want to improve index statistics handling, this is a necessary prior step.